### PR TITLE
Migrate to Postgres in Development

### LINKED ISSUE #180: Use Postgres in Development
Replace the sqlite3 development database with Postgres to better represent the production environment.

### PR DESCRIPTION
**Pull request by shikharish**
Closes #180

### CODE REVIEW

File: tests/.env
Comment (harshkhandeparkar) [inline tests/.env:1]: Is this supposed to be committed? If it is a template, rename it it `.env.template` and gitignore the `.env`.

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+services:
+  database:
```
Comment (harshkhandeparkar) [inline docker-compose.yaml:2]: Is a compose file required for a single service?

File: tests/docker-compose.test.yaml
Comment (harshkhandeparkar) [inline tests/docker-compose.test.yaml:1]: Is this file the same as the compose file in the root directory? If yes, can one of them reused?

File: utils/database.go [hunk 2]
```diff
@@ -9,7 +9,6 @@ import (
 	"github.com/kossiitkgp/kwoc-backend/v2/models"
 
 	"gorm.io/driver/postgres"
-	"gorm.io/driver/sqlite"
```
Comment (harshkhandeparkar) [inline utils/database.go:12]: Was this removed from the go mod file? Run `go mod tidy` if not.

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: tests/.env
Comment (shikharish) [reply tests/.env:1]: It is a test env file. Have updated `env.template` too.

File: docker-compose.yaml (hunk 1 above)
Comment (shikharish) [reply docker-compose.yaml:2]: Wont we need one for the volume?

File: tests/docker-compose.test.yaml
Comment (shikharish) [reply tests/docker-compose.test.yaml:1]: Yes, I made it first thinking there might me some differences. Will use the one in the root directory only.

File: tests/.env
Comment (harshkhandeparkar) [reply tests/.env:1]: Is it exactly the same for each test? If yes, we can hardcode the information in the workflow env vars itself.

File: docker-compose.yaml (hunk 1 above)
Comment (harshkhandeparkar) [reply docker-compose.yaml:2]: Is the volume required?

File: tests/.env
Comment (shikharish) [reply tests/.env:1]: Yes can do that.

File: docker-compose.yaml (hunk 1 above)
Comment (shikharish) [reply docker-compose.yaml:2]: I believe so, for the postgres database.

Comment (harshkhandeparkar) [reply docker-compose.yaml:2]: Is the volume used anywhere else? If not, it's doing nothing.

Comment (harshkhandeparkar) [reply docker-compose.yaml:2]: A volume is used for sharing storage between containers

Comment (shikharish) [reply docker-compose.yaml:2]: Ohkay. But there is already a Dockerfile in the root directory.

Comment (harshkhandeparkar) [reply docker-compose.yaml:2]: Hmm, fine.

File: README.md [hunk 3]
```diff
@@ -64,10 +64,10 @@
 ### Setting Up Locally
 
 - Install [Go](https://go.dev/).
+- Install [Docker](https://docs.docker.com/engine/install/).
```
Comment (harshkhandeparkar) [inline README.md:67]: Can you also add the steps required to set up the db locally?

Comment (harshkhandeparkar) [reply README.md:67]: Also replace SQLiteStudio with a postgres equivalent.

File: tests/common_test.go [hunk 4]
```diff
@@ -53,18 +55,18 @@ func expectResponseJSONBodyToBe[T comparable](t *testing.T, res *httptest.Respon
 }
 
 func setTestDB() *gorm.DB {
-	os.Setenv("DEV", "true")
-	os.Setenv("DEV_DB_PATH", "testDB.db")
+	_ = godotenv.Load("../.env")
```
Comment (harshkhandeparkar) [inline tests/common_test.go:58]: Is this required anymore?

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (shikharish) [reply README.md:67]: > Also replace SQLiteStudio with a postgres equivalent.
Not really sure what would be a proper equivalent. Can find several options on google.

File: tests/common_test.go (hunk 4 above)
Comment (shikharish) [reply tests/common_test.go:58]: Yes, we would have to load the env file as gorm needs the env variables.

File: README.md (hunk 3 above)
Comment (harshkhandeparkar) [reply README.md:67]: You can list multiple options. They have to be free and open source and free of cost, though. Being easy to install and use would be a nice to have.

Comment (harshkhandeparkar) [reply README.md:67]: A desktop app would be best since it will have minimal setup.

File: tests/common_test.go (hunk 4 above)
Comment (harshkhandeparkar) [reply tests/common_test.go:58]: For testing, aren't the env variables directly added to he environment? Or is the `.env` required for local testing?

Comment (harshkhandeparkar) [reply tests/common_test.go:58]: Well I guess it can't hurt so it's fine.

Comment (shikharish) [reply tests/common_test.go:58]: Yes, .env file is required for local testing.

File: README.md [hunk 5]
```diff
@@ -64,13 +64,14 @@
...
 - Clone this repository.
 - Run `go mod download` in the cloned repository to download all the dependencies.
 - Create a `.env` file to store all the [environment variables](#environment-variables). You can use the `.env.template` file for this.
-- Set the `DEV` environment variable to `true`.
+- Run `docker compose up -d` to start the PostgreSQL database container. (Make sure all the required [environment variables](#environment-variables) are set)
 - Optionally set up [Github OAuth](#github-oauth) to test the endpoints which require login. (See also: [Endpoints](#endpoints))
 - Run `go run cmd/backend.go` to start the server.
-- Optionally install [SQLiteStudio](https://sqlitestudio.pl/) or a similar tool to help manage the local database `devDB.db` (sqlite3).
+- Optionally install [pgAdmin](https://www.pgadmin.org/) or [DBeaver](https://dbeaver.io/) or a similar tool to help manage the local database `devDB.db` (postgresql).
```
Comment (harshkhandeparkar) [inline README.md:74]: `devDB.db` is no longer used :)

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -64,13 +64,14 @@
 ### Setting Up Locally
 
 - Install [Go](https://go.dev/).
+- Install [Docker](https://docs.docker.com/engine/install/).
```
Comment (harshkhandeparkar) [inline README.md:67]: Does docker compose get installed along with this or does it have to be installed as a separate component?

Comment (shikharish) [reply README.md:67]: It does get installed with docker engine.

**Review by harshkhandeparkar — APPROVED** [review]

**Comment by harshkhandeparkar** [comment]
@shikharish please rebase.

**Review by harshkhandeparkar — APPROVED** [review]

**Comment by rajivharlalka** [comment]
LGTM , can merge :shipit:

**Review by rajivharlalka — APPROVED** [review]

**Comment by harshkhandeparkar** [comment]
Thank you @shikharish! :rocket:

**Pull request merged**